controllers/ovs/flowsconfig.go

```diff
@@ -0,0 +1,144 @@
+// the target FlowCollector ipfix section map
+func (c *FlowsConfigController) Reconcile(
+	ctx context.Context, target *flowsv1alpha1.FlowCollector) error {
+	rlog := log.FromContext(ctx, "component", "FlowsConfigController")
+
+	if !target.ObjectMeta.DeletionTimestamp.IsZero() {
+		rlog.Info("no need to reconcile status of a FlowCollector that is being deleted. Ignoring")
+		return nil
+	}
+	current, err := c.current(ctx)
+	if err != nil {
+		return err
+	}
+	desired, err := c.desired(ctx, target)
+	// compare current and desired
+	if err != nil {
+		return err
+	}
+
+	if current == nil {
+		rlog.Info("Provided IPFIX configuration. Creating " + ovsConfigName + " ConfigMap")
+		return c.client.Create(ctx, c.flowsConfigMap(desired))
+	}
+
+	if desired != current {
+		rlog.Info("Provided IPFIX configuration differs current configuration. Updating")
+		return c.client.Update(ctx, c.flowsConfigMap(desired))
+	}
+
+	rlog.Info("No changes needed")
+	return nil
+}
+
+func (c *FlowsConfigController) Finalize(ctx context.Context) error {
+	err := c.client.DeleteAllOf(ctx, &corev1.ConfigMap{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      ovsConfigName,
+			Namespace: c.operatorNamespace,
+		},
+	}, client.InNamespace(c.operatorNamespace),
+		client.MatchingFields{"metadata.name": ovsConfigName})
+
+	return err
+}
+
+func (c *FlowsConfigController) current(ctx context.Context) (*flowsConfig, error) {
+	curr := &corev1.ConfigMap{}
+	if err := c.client.Get(ctx, types.NamespacedName{
+		Name:      ovsConfigName,
+		Namespace: c.operatorNamespace,
+	}, curr); err != nil {
+		if errors.IsNotFound(err) {
+			// the map is not yet created. As it is associated to a flowCollector that already
+			// exists (premise to invoke this controller). We will handle accordingly this "nil"
+			// as an expected value
+			return nil, nil
+		}
+		return nil, fmt.Errorf("retrieving %s/%s configmap: %w",
+			c.operatorNamespace, ovsConfigName, err)
+	}
+
+	return configFromMap(curr.Data)
+}
+
+func (c *FlowsConfigController) desired(
+	ctx context.Context, coll *flowsv1alpha1.FlowCollector) (*flowsConfig, error) {
+
+	conf := flowsConfig{FlowCollectorIPFIX: coll.Spec.IPFIX}
+
+	// According to the "OVS flow export configuration" RFE:
+	// nodePort be set by the NOO when the collector is deployed as a DaemonSet
+	// sharedTarget set when deployed as Deployment + Service
+	switch coll.Spec.GoflowKube.Kind {
+	case constants.DaemonSetKind:
+		conf.NodePort = coll.Spec.GoflowKube.Port
+	case constants.DeploymentKind:
+		svc := corev1.Service{}
+		if err := c.client.Get(ctx, types.NamespacedName{
+			Namespace: c.operatorNamespace,
+			Name:      constants.GoflowKubeName,
+		}, &svc); err != nil {
+			return nil, err
+		}
+		// TODO: if spec/goflowkube is empty or port is empty, fetch first UDP port in the service spec
+		conf.SharedTarget = fmt.Sprintf("%s:%d", svc.Name, coll.Spec.GoflowKube.Port)
```

Comment:
the service ClusterIP is necessary here, as OVS won't resolve domain names

Reply:
However I wonder if the cluster IP is immediately available after service creation or if we need to introduce some delay / retries ...

Reply:
If it is not available, we can return a status error asking for a retry

Reply:
Thinking more about it, I think we should implement the DNS resolution in the CNO (task [NETOBSERV-66](https://issues.redhat.com/browse/NETOBSERV-66) ), as this limitation is an internal detail from OVS.

Reply:
ok, sounds good to me

Reply:
Update: CNO DNS configuration prevents proper/simple DNS resolution and the RFE spec states that we should provide the IP already resolved, so I have implemented the DNS resolution already in this PR ([commit](https://github.com/netobserv/network-observability-operator/pull/8/commits/4d9e9b187c5e37e6b661ccffc8658f71f5e02335))